corehq/apps/users/models.py

```diff
@@ -1764,17 +1764,21 @@ def get_language_code(self):
     def location(self):
         from corehq.apps.locations.models import Location
         if self.location_id:
-            return Location.get(self.location_id)
-        else:
-            return None
+            try:
+                return Location.get(self.location_id)
+            except ResourceNotFound:
+                self.unset_location()
```

Comment:
If a location is archived, this method and property will still return the (archived) location - they don't respect the flag.  I'm not actually sure how locations can get deleted, my guess is it's a leftover from that data cleanup we did a while back.